### ISSUE TITLE
Misleading error message when returning 'this' by reference from class member

### ISSUE THREAD

**Issue by akarpov89** [dotnet/roslyn#17707]
**Version Used**: 
Master branch @ March 9
**Steps to Reproduce**:

Compile the following code:
```csharp
class C {
    ref C M() => ref this;   
}
```

**Expected Behavior**:
The compiler emits error "Class members cannot return 'this' by reference".

**Actual Behavior**:
The compiler emits error "Struct members cannot return 'this' or other instance members by reference"

**Comment by akarpov89** [comment]
It is also possible to use message "Cannot return 'this' by reference" without specifying if the member belongs to class or struct.

**Comment by alrz** [comment]
Dup of #15704 and already has a fix here: https://github.com/dotnet/roslyn/pull/15706
